setools/terulequery.py

```diff
@@ -162,13 +165,21 @@ def results(self):
             #
             # Matching on target type
             #
-            if self.target and not match_indirect_regex(
+            if self.target and not self.target_self and not \
+                    match_indirect_regex(
                     rule.target,
                     self.target,
                     self.target_indirect,
                     self.target_regex):
                 continue
 
+            if self.target_self and not match_indirect_regex(
+                    rule.target,
+                    rule.source,
+                    False,
+                    False):
+                continue
```

Comment:
Please use explicit `rule.target == rule.source` check here so it is crystal clear.